ini/parse.ts

```diff
@@ -81,7 +142,70 @@ export interface ParseOptions {
  */
 export function parse<T extends object>(
   text: string,
-  options?: ParseOptions,
+  options: ParseOptions = {},
 ): T {
-  return IniMap.from(text, options).toObject<T>();
+  if (typeof text !== "string") {
+    throw new SyntaxError(`Unexpected token ${text} in INI at line 0`);
+  }
+
+  const { reviver = defaultReviver } = options;
+
+  const root = {} as T;
+  let object: object = root;
+  let sectionName: string | undefined;
+
+  let lineNumber = 0;
+  for (let line of readTextLines(text)) {
+    line = line.trim();
+    lineNumber += 1;
+
+    // skip empty lines
+    if (line === "") continue;
+
+    // skip comment
+    if (isComment(line)) continue;
+
+    if (isSection(line, lineNumber)) {
+      sectionName = SECTION_REGEXP.exec(line)?.groups?.name;
+
+      if (!sectionName) {
+        throw new SyntaxError(
+          `Unexpected empty section name at line ${lineNumber}`,
+        );
+      }
```

Comment:
This seems allowing empty section name now. `parse("[ ]\na=1")` currently throws, but now it returns `{ " ": { a: 1 } }`

Reply:
I fixed the regexp to allow `\S+` as section name instead of `.+`.